modules/monitoring-exposing-custom-application-metrics-for-horizontal-pod-autoscaling.adoc

```diff
@@ -0,0 +1,227 @@
+// Module included in the following assemblies:
+//
+// monitoring/monitoring.adoc
+
+[id='exposing-custom-application-metrics-for-horizontal-pod-autoscaling-{context}']
+= Exposing custom application metrics for horizontal pod autoscaling
+
+You can use the `prometheus-adapter` resource to expose custom application metrics for the horizontal pod autoscaler.
+
+.Prerequisites
+
+* Make sure you configured the cluster for application monitoring. In this example, it is presumed that Prometheus and Alertmanager instances were installed in the `default` namespace.
+* Make sure you configured monitoring for your application monitoring. In this example, it is presumed that the application and the service monitor for it were installed in the `default` namespace.
+
+.Procedure
+
+. Create a YAML file for your configuration. In this example, it is called `deploy.yaml`.
+
+. Add configuration for creating the service account, necessary roles, and role bindings for `prometheus-adapter`:
++
+[source,yaml,subs=quotes]
+  kind: ServiceAccount
```

Comment:
two spaces at the beginning of each line, if customer copy the file, they need to delete the two spaces before each line